pkg/alertmanager/matcher.go

```diff
@@ -0,0 +1,94 @@
+package alertmanager
+
+import (
+	"reflect"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/alertmanager/matchers/parse"
+	"github.com/prometheus/alertmanager/pkg/labels"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ParseMetrics struct {
+	Total             prometheus.Counter
+	DisagreeTotal     prometheus.Counter
+	InvalidTotal      prometheus.Counter
+	IncompatibleTotal prometheus.Counter
+}
+
+func NewParseMetrics(r prometheus.Registerer) *ParseMetrics {
```

Comment:
These are the metrics I thought of that will let us understand:

1. How many tenants have matchers that are incompatible with the UTF-8 parser
2. How many tenants have matchers that parse in both parsers, but produce different parsings. Some false positives here are expected, for example, tenants who are attempting to use hex literals to encode UTF-8 code points.
3. I also added a metric to see if any tenants somehow have invalid matchers, which shouldn't happen?